app/controllers/posts_controller.rb

```diff
@@ -184,6 +184,7 @@ def reply_history
 
   def destroy
     post = find_post_from_params
+    RateLimiter.new(current_user, "delete_post", 3, 1.minute).performed! unless current_user.staff?
```

Comment:
I don't think this should be handled at the controller level but rather in [PostDestroyed#destroy](https://github.com/discourse/discourse/blob/master/lib/post_destroyer.rb). 

Also, I think we should have a SiteSettings for the cooldown time.